static/src/javascripts/projects/common/modules/commercial/build-page-targeting.ts

```diff
@@ -346,70 +331,89 @@ const filterEmptyValues = (pageTargets: Record<string, unknown>) => {
 	return filtered;
 };
 
-const rebuildPageTargeting = () => {
-	latestCmpHasInitialised = cmp.hasInitialised();
-	const adConsentState = getAdConsentFromState(latestCMPState);
-	const ccpaState = latestCMPState?.ccpa
-		? latestCMPState.ccpa.doNotSell
-		: null;
-	const tcfv2EventStatus = latestCMPState?.tcfv2
-		? latestCMPState.tcfv2.eventStatus
-		: 'na';
+const getConsentRelatedPageTargeting = (
+	canTargetAds: boolean,
+	consentState?: ConsentState,
+): PageTargeting => {
+	const amtgrp = consentState?.tcfv2
+		? getAdManagerGroup(canTargetAds)
+		: getAdManagerGroup();
+
+	if (!consentState) {
+		return {
+			amtgrp,
+			cmp_interaction: 'na',
+			consent_tcfv2: 'na',
+			pa: 'f',
+			rdp: 'na',
+		};
+	}
+	return {
+		amtgrp,
+		cmp_interaction: consentState.tcfv2
+			? consentState.tcfv2.eventStatus
+			: 'na',
+		consent_tcfv2: consentState.tcfv2
+			? consentedToAllPurposes(consentState.tcfv2.consents)
+				? 't'
+				: 'f'
+			: 'na',
+		pa: canTargetAds ? 't' : 'f',
+		rdp: consentState.ccpa
+			? consentState.ccpa.doNotSell
+				? 't'
+				: 'f'
+			: 'na',
+	};
+};
 
+const getPageTargeting = (consentState?: ConsentState): PageTargeting => {
```

Comment:
Consent state is still optional as I want to test this fully (including appNexus bidders) first and then once I'm happy I will make it mandatory.

Reply:
It will stay optional for now for reasons explained [here](https://github.com/guardian/frontend/pull/24498#discussion_r797548260).